### PR TITLE
Add docs to expectations methods

### PR DESCRIPTION
**Pull request by straight-shoota**

### CODE REVIEW

File: src/spec/expectations.cr [hunk 1]
```diff
@@ -185,68 +185,90 @@ module Spec
...
     end
 
+    # Creates an `Expectation` that passes if actual is within *delta* of *expected*.
     def be_close(expected, delta)
       Spec::CloseExpectation.new(expected, delta)
     end
 
+    # Returns a factory to create a comparison `Expectation` that:
+    #
+    # * passes if actual is lesser than *value*: `be.<(value)`
```
Comment (RX14) [inline src/spec/expectations.cr]: Perhaps using the longer `be < value` is clearer here?

File: src/spec/expectations.cr [hunk 2]
```diff
@@ -185,68 +185,90 @@ module Spec
...
     def contain(expected)
       Spec::ContainExpectation.new(expected)
     end
 
+    # Creates an `Expectation` that passes if actual is of type *type* (`is_a?`).
     macro be_a(type)
       Spec::BeAExpectation({{type}}).new
     end
 
+    # Creates an `Expectation` that passes if the block raises an exception.
```
Comment (RX14) [inline src/spec/expectations.cr]: These `expect_*` macros don't create an expectation, they raise themselves.

File: src/spec/expectations.cr (hunk 1 above)
Comment (straight-shoota) [reply src/spec/expectations.cr]: Yes. For some reason I assumed the `.` was require (dunno why, something failed when I was experimenting with it).

File: src/spec/expectations.cr [hunk 3]
```diff
@@ -185,68 +185,90 @@ module Spec
...
     def contain(expected)
       Spec::ContainExpectation.new(expected)
     end
 
+    # Creates an `Expectation` that passes if actual is of type *type* (`is_a?`).
     macro be_a(type)
       Spec::BeAExpectation({{type}}).new
     end
 
+    # Runs the block and passes if it raises an exception.
```
Comment (yxhuvud) [inline src/spec/expectations.cr]: Worth noting about this method is that this variant that doesn't verify it against an actual exception type has been deprecated in rspec. From the places where I fixed the deprecation errors for that, I can say that the reason for the deprecation was very motivated, as there was lots of hidden errors and unexpected exceptions being caught.

**Comment by straight-shoota** [comment]
@RX14 Yes, there is still a lot missing. I'll try to condense some of the testing guide to a API doc for `Spec`.

File: src/spec/expectations.cr (hunk 3 above)
Comment (straight-shoota) [reply src/spec/expectations.cr]: Good point! I'll add a note about this and suggest removing the matcher in a separate issue.

**Comment by RX14** [comment]
@straight-shoota just "Expectations are used by `Spec::ObjectExtensions#should` and `Spec::ObjectExtensions#should_not`." would be enough clarification.

**Comment by straight-shoota** [comment]
@RX14 The module `Spec` itself is already documented, that's what I used as a starting point for the testing guide.
I'll try to add documentation to the methods.

**Comment by RX14** [comment]
Needs a rebase.

**Comment by straight-shoota** [comment]
Ready to merge! 💯

**Comment by Sija** [comment]
Ready to 🎲 !

**Pull request merged**